irods/client_configuration/__init__.py

```diff
@@ -111,7 +126,38 @@ def save(root = None, string='', file = ''):
         if _file and auto_close_settings:
             _file.close()
 
-def _load_config_line(root, setting, value):
+@contextlib.contextmanager
+def loadlines(entries, common_root = sys.modules[__name__]):
+    root_item = [('root',common_root)]
+    entries_ = copy.deepcopy(entries)
+    identity = lambda _:_
+    try:
+        # Load config values.
+        entries_ = []
+        for e in entries:
+            e_ = dict(root_item + list(e.items()))
+            L = []
+            _load_config_line(eval_func = identity, return_old = L, **e_)
+            e_['value'] = L[0]
+            entries_.append(e_)
+        yield
+    finally:
+        # Restore old values.
+        for e_ in entries_:
+            _load_config_line(eval_func = identity, **e_)
+
+
+# TODO remove this test
+def test_it():
+    with loadlines(entries=(
+        dict(setting='data_objects.auto_close',value=True),
+    )):
+        print('dac (during) =',data_objects.auto_close)
+        pass
+    print('dac (after) =',data_objects.auto_close)
+    1
+
+def _load_config_line(root, setting, value, return_old=(), eval_func = ast.literal_eval):
```

Comment:
Why is the default value for `return_old` an empty tuple? Really asking, little-to-no rhetoric here. :)

Reply:
I don't know if `( )` is better than `None` here, tbh.  It's in keeping with a practice I have relied upon for signalling to the called function whether to pass back a value or not.  In this case, the arg is a tuple (the default being an empty one), the caller isn't interested in receiving the old value.  None would serve as well, but the tuple is "more like" the empty list and therefore I tend to use it in the place of `None`.  If it's confusing, I can use the latter.

Reply:
Okay. I'll have to think on that.

I think I tend to feel that `None` with a detailed docstring would be better, but that might be personal preference. Better still might be splitting up the function and having two things with different return values that call the same underlying logic (if it's possible).

Reply:
Done, switched to None as default for `return_old` .